bCNC/Pendant.py

```diff
@@ -112,9 +112,11 @@ def do_GET(self):
 		elif page == "/icon":
 			if arg is None: return
 			filename = os.path.join(iconpath, arg["name"]+".gif")
-			self.do_HEAD(200, content="image/gif", cl=os.path.getsize(filename))
 			try:
 				f = open(filename,"rb")
+				f.seek(0,2)
```

Comment:
why are you calling the seek() ?

Reply:
Hi!
Sorry, but I don't remember precisely why I placed this seek() here. I think without this seek() it starts to read file somewhere at different location, but not at beginning of the file. One thing I can surely say - it was on Debian Linux 10 (buster) and python 3. I think you can try to remove this line and check how things will work.

Reply:
According to https://www.geeksforgeeks.org/python-seek-function/
`f.seek(0,2)` does jump to the end of file. this makes me slightly suspicious about this PR :-)

Reply:
Oh, now I remember. There were problem with os.path.getsize() - IDK why, but it returning wrong value.
So this is workaround - seek to the end of file and get current position to determine file length.

Reply:
Ok, makes sense... Can you please put this explaination in form of code comments?

Reply:
It's occuring in several places. perhaps preferred way imho would be to create separate file length function that would:

1. store current position
2. seek to end and store file length to variable
3. seek back to initial position
4. return file length
5. contain comments about why we do this

Reply:
I thought you already included this changes (using something different method than merge pull) after 27 Jul 2023.
If no - well, may be this week I will have some time to do it.